### PR TITLE
feature: introduce CI

### PR DESCRIPTION
**Pull request by boesing**
|    Q          |   A
|-------------- | ------
| New Feature   | yes
| QA            | yes

### Description

Initial work on adding CI pipeline for the matrix action.

### CODE REVIEW

**Comment by boesing** [comment]
I see, we need a slightly changed container so it does not always do the composer magic...

**Comment by boesing** [comment]
@weierophinney Any chance that we get this merged?
I've added plenty of tests already and would be happy to take this as a base for #83 so we can finally ship that whole thing.

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    name: Generate job matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.matrix.outputs.matrix }}
+    steps:
+      - name: Checkout sourcecode
+        uses: actions/checkout@v3
+      - name: Gather integration test directory names
+        id: matrix
+        run: cd tests; echo "::set-output name=matrix::[\"$(ls -d * | tr '\n' ' ' | sed 's/ $//' | sed 's/ /\",\"/g')\"]"
```
Comment (Ocramius) [inline .github/workflows/continuous-integration.yml:26]: Maybe a bit cryptic - TBH, a few comments in this file would clarify that we're running a bunch of parallel tests to check that the action will generate the right JSON for the various directories in here.

That said, this is 💯 

Awesome!

**Review by Ocramius — APPROVED** [review]
IMO 🚢

Possibly worth merging this one before #83, to see if we are at parity too.

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (boesing) [reply .github/workflows/continuous-integration.yml:26]: True. Will add this as a comment. Will merge this afterwards and then rebase #83

**Pull request merged**